lib/sla.js

```diff
@@ -45,80 +48,135 @@ function designateMixingBridge(client, bridgeName) {
 }
 
 /**
- * Adds inbound and originated channels to the bridge.
+ * Updates the state of the shared extension.
  * @param {Object} client - Object that contains information from the ARI
  *   connection.
- * @param {Object} inbound - Incoming channel to be added to the bridge.
+ * @param {String} name - name of the shared extension to update.
+ * @param {String} state - state to update the shared extension to.
+ */
+function updateState(client, name, state) {
+  var deviceState = Q.denodeify(client.deviceStates.update.bind(client));
+  deviceState({deviceName: 'Stasis:' + name, deviceState: state});
+}
+
+/**
+ * Accesses the current state of the shared extension.
+ * @param {Object} client - Object that contains information from the ARI
+ *   connection.  Houses the device states
+ * @param {String} name - name of the shared extension to access.
+ * @return {String} - the current device state of this extension
+ */
+function getState(client, name) {
+  var getDeviceState = Q.denodeify(client.deviceStates.get.bind(client
+        .deviceStates));
+  return getDeviceState({deviceName: 'Stasis:' + name}).then(function (ds) {
+    return ds.state;
+  });
+}
+
+/**
+ * Adds the caller channel and originated channel (if applicable) to the bridge.
+ * @param {Object} client - the client (used for hanging up at the end).
+ * @param {Object} channel - Incoming channel to be added to the bridge.
  * @param {Object} dialed - Created/dialed channel to be added as well
  * @param {Object} bridge - Bridge that channels are to be added to.
+ * @param {Object} extension - the object data related to this sharedExtension
  * @return {Q} Q promise object.
  */
-function addChannelsToBridge(client, inbound, dialed, bridge) {
+function addChannelsToBridge(client, channel, dialed, bridge, extension) {
+  // The only way that the channel length will be 1 is if the outside user is
+  // still inside of the shared extension waiting for others to answer.
+  if (bridge.channels.length === 1) {
+    bridge.stopMoh();
+    updateState(client, bridge.name, inUse);
+  }
+
+  // Resets listeners, so that redundant events aren't accepted.
+  bridge.removeAllListeners('ChannelEnteredBridge');
+  bridge.removeAllListeners('ChannelLeftBridge');
   var defer = Q.defer();
-  var numInBridge = 0;
 
   console.log('Adding channels to bridge %s', bridge.id);
 
   var addChannel = Q.denodeify(bridge.addChannel.bind(bridge));
-  addChannel({channel: [inbound.id, dialed.id]})
+  addChannel({channel: channel.id})
     .catch(function (err) {
       defer.reject(err);
     });
 
+  if (dialed) {
+    addChannel({channel: dialed.id})
+      .catch(function (err) {
+        defer.reject(err);
+      });
+  }
+  bridge.on('ChannelEnteredBridge', channelEntered);
```

Comment:
The mechanics of firing an event are relatively straightforward: the event has some collection of subscribers (callbacks) that are invoked when the event is raised. 

The event does not care if {n} callbacks are identical; it is going to treat each request for subscription as an individual and unique request. In essence, each time line 113 is executed, the event appends the callback to its list of subscribers.

What Matt is saying, is by subscribing to the event here, on line 113, rather than in some other place where the code will only be executed once, will result in the list of subscribers growing inordinately, which may have unexpected side-effects the longer the application is alive.